libr/core/cmd_search.c

```diff
@@ -230,6 +231,8 @@ R_API int r_core_search_prelude(RCore *core, ut64 from, ut64 to, const ut8 *buf,
 	int ret;
 	ut64 at;
 	ut8 *b = (ut8 *)malloc (core->blocksize);
+	int timeout = r_config_get_i (core->config, "anal.aaptimeout");
+	time_t stop_time = time(NULL) + timeout;
```

Comment:
make an api out of this: 

ut64 t = r_cons_timeout(); 
and then t = r_cons_timeout_check (t);

which may return the updated time or 0 if timeout happens. this way the function can be used for differential timeouts and such.